eclipse-theia/theia
# core: re-export `dompurify` as a shared dependency

#### Rationale

The `dompurify` dependency was previously included in five extensions, and was used to sanitize content which may cause security concerns. For these reasons, the dependency was moved up to `core` where it can be shared.

#### What it does


The pull-request includes the following updates:
- adds `dompurify` as a shared dependency from `@theia/core`
- bumps the `dompurify` dependency to the latest version (confirmed it is **license compatible**)
- suppresses the sanitized content warnings for content which is known to be explicitly handled.

#### How to test


Verify that the different extensions which previously used `dompurify` work correctly:
- console successfully renders ansi output
- notifications are successfully rendered
- the 'readme' of extensions from the 'extensions-view' are successfully rendered

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
